ui/tests/integration/components/pki/page/pki-issuer-rotate-root-test.js

```diff
@@ -0,0 +1,257 @@
+    await click(SELECTORS.toggle);
+    assert.strictEqual(findAll('[data-test-row-label]').length, 0, 'it hides again');
+
+    // customize form
+    await click(SELECTORS.customRadioSelect);
+    assert.dom(SELECTORS.generateRootForm).exists('it renders generate root form');
+    assert
+      .dom(SELECTORS.input('permittedDnsDomains'))
+      .hasValue(this.parsedRootCert.permittedDnsDomains, 'form is prefilled with values from old root');
+    await click(SELECTORS.generateRootCancel);
+    assert.ok(this.onCancel.calledOnce, 'custom form calls @onCancel passed from parent');
+    await click(SELECTORS.oldRadioSelect);
+    await click(SELECTORS.rotateRootCancel);
+    assert.ok(this.onCancel.calledTwice, 'old root settings form calls @onCancel from parent');
+
+    // validations
+    await fillIn(SELECTORS.input('commonName'), '');
+    await fillIn(SELECTORS.input('issuerName'), 'default');
+    await click(SELECTORS.rotateRootSave);
+    assert.dom(SELECTORS.validationError).hasText('There are 2 errors with this form.');
+    assert.dom(SELECTORS.input('commonName')).hasClass('has-error-border', 'common name has error border');
+    assert.dom(SELECTORS.input('issuerName')).hasClass('has-error-border', 'issuer name has error border');
+  });
+
+  test('it sends request to rotate/internal on save when using old root settings', async function (assert) {
+    assert.expect(1);
+    this.server.post(`/${this.backend}/root/rotate/internal`, () => {
+      assert.ok('request made to correct default endpoint type=internal');
+    });
+    await render(
+      hbs`
+      <Page::PkiIssuerRotateRoot
+        @oldRoot={{this.oldRoot}}
+        @newRootModel={{this.newRootModel}}
+        @breadcrumbs={{this.breadcrumbs}}
+        @onCancel={{this.onCancel}}
+        @onComplete={{this.onComplete}}
+      />
+    `,
+      { owner: this.engine }
+    );
+    await click(SELECTORS.rotateRootSave);
+  });
+
+  function testEndpoint(test, type) {
+    test(`it sends request to rotate/${type} endpoint on save with custom root settings`, async function (assert) {
+      assert.expect(1);
+      this.server.post(`/${this.backend}/root/rotate/${type}`, () => {
+        assert.ok('request is made to correct endpoint');
+      });
+      await render(
+        hbs`
+      <Page::PkiIssuerRotateRoot
+        @oldRoot={{this.oldRoot}}
+        @newRootModel={{this.newRootModel}}
+        @breadcrumbs={{this.breadcrumbs}}
+        @onCancel={{this.onCancel}}
+        @onComplete={{this.onComplete}}
+      />
+    `,
+        { owner: this.engine }
+      );
+      await click(SELECTORS.customRadioSelect);
+      await fillIn(SELECTORS.typeField, type);
+      await click(SELECTORS.generateRootSave);
+    });
+  }
+  testEndpoint(test, 'internal');
+  testEndpoint(test, 'exported');
+  testEndpoint(test, 'existing');
+  testEndpoint(test, 'kms');
+
+  test('it renders details after save for exported key type', async function (assert) {
```

Comment:
this test case technically won't happen in the wild because selecting 'use old settings' should only generate an internal key type and the <PkiGenerateRoot> child component will handle displaying the detail views for the other key types but wanted to test in case we decide to reuse `PkiInfoTableRows`

Reply:
Exported key type would happen for not `use old settings` though right?